docs/Onboarding Docs/01_Razor_Basics.md

```diff
@@ -0,0 +1,52 @@
+
+## Razor Editors: Legacy vs New
+
+| Aspect | Razor Legacy | Legacy .NET Core Razor Editor | New .NET Core Razor Editor |
+| --- | --- | --- | --- |
+| **Introduction** | Introduced with ASP.NET MVC 3. | Older Razor editor for ASP.NET Core projects. | Updated Razor editor introduced in Visual Studio 2019 version 16.8. |
+| **Usage** | Used in ASP.NET MVC and ASP.NET Web Pages applications. | Used for editing Razor views and pages in ASP.NET Core projects. | Used for editing Razor views and pages in ASP.NET Core projects. |
+| **Source code** | Closed source. | Closed source. | [Open source on GitHub](https://github.com/dotnet/razor/) |
+| **File Extensions** | `.cshtml` for C#, `.vbhtml` for VB.NET. | N/A | N/A |
+| **Functionality** | Creates dynamic web pages that combine HTML and server-side code. | Provides basic features like syntax highlighting and IntelliSense for Razor syntax. | Provides improved functionality and performance, including better IntelliSense, improved syntax highlighting, support for Razor formatting, better diagnostics, and features like "Go to Definition" and "Find All References" for Razor components and their parameters. |
+| **Support** | Still supported for maintaining existing applications. New development typically done using newer versions of Razor in ASP.NET Core. | N/A | N/A |
+| **Configuration** | N/A | To switch to the legacy editor, go to "Tools" > "Options" > "Environment" > "Preview Features" and uncheck the box that says "Use the new Razor editor for .NET Core apps". | To switch to the new editor, go to "Tools" > "Options" > "Environment" > "Preview Features" and check the box that says "Use the new Razor editor for .NET Core apps". |
+| **Implementation** | N/A | Had a more monolithic design, with all language services being provided directly by the editor. Did not use LSP or TextMate grammars, and its integration with the rest of Visual Studio was more limited. | Implemented using the Language Server Protocol (LSP), which allows it to provide language services like IntelliSense, diagnostics, and code actions. Uses TextMate grammars for syntax highlighting, providing consistent and accurate highlighting for Razor, C#, and HTML. Integrated with the rest of Visual Studio through the editor API, allowing it to provide features like "Go to Definition", "Find All References", and code actions that work seamlessly with the rest of the IDE. Includes special support for Blazor. |
+
+## Razor Support Across ASP.NET Versions
+
+Different versions of ASP.NET support different features of Razor. Here's a summary:
+
+| TFM | Razor Support |
+| --- | --- |
+| **.NET Framework (<= 4.8)** | Supports Razor syntax with C# and VB.NET. Used in ASP.NET MVC and ASP.NET Web Pages applications. |
+| **.NET Core 1.x - 3.1** | Supports Razor syntax with C# only. Used in ASP.NET Core MVC and Razor Pages applications. |
```

Comment:
IIRC Blazor was in preview at this time too which was supported then. Don't know if it's worth calling out

Reply:
```suggestion
| **.NET Core 1.x - 3.1** | Supports Razor syntax with C# only. Used in ASP.NET Core MVC, Razor Pages applications, and had preview support for Blazor. |
```